ci/local/README.md

```diff
@@ -0,0 +1,44 @@
+## Purpose
+
+This script is designed for developer and contributor use. This tool mimics the actions of gpuCI on your local machine. This allows you to test and even debug your code inside a gpuCI base container before pushing your code as a GitHub commit.
+
+
+## Usage
+
+```
+bash build.sh [-h] [-H] [-s] [-r <repo_dir>] [-i <image_name>]
+Build and test your local repository using a base gpuCI Docker image
+
+where:
+    -H   Show this help text
+    -r   Path to repository (defaults to working directory)
+    -i   Use Docker image (default is gpuci/rapidsai-base:cuda10.0-ubuntu16.04-gcc5-py3.6)
+    -s   Skip building and testing and start an interactive shell in a container of the Docker image
+```
+
+Example Usage:
+`bash build.sh -r ~/rapids/cudf -i gpuci/cuda9.2-ubuntu16.04-gcc5-py3.6`
```

Comment:
Is there an option to specify a git repo and branch to pull and build/test? I realize you can do that by checking it out locally and then running the script, but it might be helpful if I'm asked to help triage failures in someone else's PR or branch, for example.

Reply:
+1

Reply:
I've thought about your comment and I understand why you would want this. However I believe that functionality may be outside the scope of this script. I think what you're asking for would be more easily implemented as a small tool outside the repo itself (with this new tool calling this script in it after git cloning). That is just my viewpoint from a design perspective. Open to your opinions on this.

Reply:
I see @dillon-cullinan 's point, but I'm also wondering if we've scoped this a little too narrow. It wouldn't be hard to add what @harrism and @kkraus14 are asking for, and maybe we just move this script outside of the repo(s) (into a "tools" repo, for example), either now or in a separate PR.

On second though, maybe we just get this script merged now and address this later. I suspect a script this useful will get several more feature requests soon, and we'll be adding functionality often.

Reply:
I agree this decision can be made in a separate issue and implemented in a separate PR.

Reply:
@rlratzel I think the main difficulty comes from the fact that some of the repos are a little different in how they are built (mainly due to repo file structure). I think we can use this as a sort of helper to higher level ones that can do what @harrism and @kkraus14 are suggesting.

I'll keep this in mind when expanding functionality.